test/ecto/query/builder_test.exs

```diff
@@ -27,6 +27,9 @@ defmodule Ecto.Query.BuilderTest do
     assert {Macro.escape(quote do type(&0.y + &1.z, :decimal) end), []} ==
            escape(quote do type(x.y + y.z, :decimal) end, [x: 0, y: 1], __ENV__)
 
+    assert {Macro.escape(quote do type(&0.y, :decimal) end), []} ==
+          escape(quote do type(field(x, :y), :decimal) end, [x: 0], __ENV__)
```

Comment:
Shouldn't we test this with an alias? I think we have a Custom.Permalink type thingy.

Reply:
I can add this as well, but even this didn't work previously.

Reply:
Let's add it because otherwise I can remove lines of code from your commit and all tests would still pass. :)

Reply:
Ping. :)

Reply:
Ping? :)

Reply:
Sorry for the delay. I've not much time at hand at the moment and I was getting into trouble where the coe would return an alias in AST format, but `Macro.escape` in the test would either escape the AST or have the module name as atom.